torch/_export/constraints.py

```diff
@@ -0,0 +1,36 @@
+from typing import Optional
+
+import torch._dynamo
+from torch._dynamo.comptime import comptime
+from torch.fx.experimental.symbolic_shapes import constrain_range
+
+
+# TODO: we want to hide this min/max stuff under some abstraction similar to
+# DynamicDim
+def add_inline_constraint(symbol, min: Optional[int] = None, max: Optional[int] = None):
+    """
+    Add min/max constraint on the intermediate symbol at tracing time
+    """
+
+    if torch._dynamo.is_compiling():
+        @comptime
+        def _(ctx):
+            node = ctx.get_local("symbol").as_proxy().node
+            min = ctx.get_local("min").as_python_constant()
+            max = ctx.get_local("max").as_python_constant()
+            constrain_range(node.meta["example_value"], min=min, max=max)
```

Comment:
Curious, mostly for my education: Why is there a double call here (vs. the one below)? Is `example_value` the concrete value of the symbol? Do we have such a thing for an unbacked symint?

Reply:
Oh sorry, you're right that we don't need the first call -- I forgot that the 2nd line calls the function.